frontend/javascripts/admin/dataset/dataset_add_wk_connect_view.js

```diff
@@ -0,0 +1,144 @@
+import {
+  CardContainer,
+  DatasetNameFormItem,
+  DatastoreFormItem,
+} from "admin/dataset/dataset_components";
+
+const FormItem = Form.Item;
+
+type OwnProps = {|
+  datastores: Array<APIDataStore>,
+  withoutCard?: boolean,
+  onAdded: (string, string) => void,
+|};
+type StateProps = {|
+  activeUser: ?APIUser,
+|};
+type Props = {| ...OwnProps, ...StateProps |};
+type PropsWithForm = {|
+  ...Props,
+  form: Object,
+|};
+
+class DatasetAddWkConnectView extends React.PureComponent<PropsWithForm> {
+  validateAndParseUrl(url: string) {
+    const delimiterIndex = url.indexOf("#!");
+    if (delimiterIndex < 0) {
+      throw new Error("The URL doesn't contain the #! delimiter. Please insert the full URL.");
+    }
+
+    const jsonConfig = url.slice(delimiterIndex + 2);
+    // This will throw an error if the URL did not contain valid JSON. The error will be handled by the caller.
+    const config = JSON.parse(decodeURIComponent(jsonConfig));
+    config.layers.forEach(layer => {
+      if (!layer.source.startsWith("precomputed://")) {
+        throw new Error(
+          "This dataset contains layers that are not supported by wk-connect. wk-connect supports only 'precomputed://' neuroglancer layers.",
+        );
+      }
+    });
+    return config;
+  }
+
+  handleSubmit = evt => {
+    evt.preventDefault();
+    const { activeUser } = this.props;
+
+    this.props.form.validateFields(async (err, formValues) => {
+      if (!err && activeUser != null) {
+        const neuroglancerConfig = this.validateAndParseUrl(formValues.url);
+        const fullLayers = _.keyBy(neuroglancerConfig.layers, "name");
+        // Remove unnecessary attributes of the layer, the precomputed source prefix needs to be removed as well
+        const layers = _.mapValues(fullLayers, ({ source, type }) => ({
+          type,
+          source: source.replace(/^(precomputed:\/\/)/, ""),
+        }));
+
+        const datasetConfig = {
+          neuroglancer: {
+            [activeUser.organization]: {
+              [formValues.name]: {
+                layers,
+              },
+            },
+          },
+        };
+
+        await addWkConnectDataset(formValues.datastore, datasetConfig);
+
+        Toast.success(messages["dataset.add_success"]);
+        trackAction("Add remote dataset");
+        await Utils.sleep(3000); // wait for 3 seconds so the server can catch up / do its thing
+        this.props.onAdded(activeUser.organization, formValues.name);
+      }
+    });
+  };
+
+  render() {
+    const { form, activeUser, withoutCard, datastores } = this.props;
+    const { getFieldDecorator } = form;
+
+    return (
+      <div style={{ padding: 5 }}>
+        <CardContainer withoutCard={withoutCard} title="Add wk-connect Dataset">
+          Currently wk-connect supports adding Neuroglancer datasets. Simply set a dataset name,
+          select the wk-connect datastore and paste the URL to the Neuroglancer dataset.
```

Comment:
In which cases would we have more than one wkc datastore here? Is this a usecase at all? It seems a bit "complicated" to me, that the user has to deal with it at all, because most of the time they won't care.
However, it's fine for now, I think. We can always streamline this later :)

Reply:
I wasn't sure, but you're probably right, there's either one or none wk-connect datastores I'd say. Do you agree @normanrz?